cpp/include/raft/mdarray.hpp

```diff
@@ -642,9 +721,111 @@ auto make_device_vector(size_t n, rmm::cuda_stream_view stream)
  * @param[in] n number of elements in vector
  * @return raft::device_vector
  */
-template <typename ElementType>
+template <typename ElementType, typename LayoutPolicy = layout_c_contiguous>
 auto make_device_vector(raft::handle_t const& handle, size_t n)
 {
-  return make_device_vector<ElementType>(n, handle.get_stream());
+  return make_device_vector<ElementType, LayoutPolicy>(n, handle.get_stream());
+}
+
+template <typename host_mdspan_type,
+          std::enable_if_t<is_host_mdspan_v<host_mdspan_type>>* = nullptr>
+auto flatten(host_mdspan_type h_mds)
+{
+  RAFT_EXPECTS(h_mds.is_contiguous(), "Input must be contiguous.");
+
+  return make_host_vector_view<typename host_mdspan_type::element_type,
+                               typename host_mdspan_type::layout_type>(h_mds.data(), h_mds.size());
+}
+
+template <typename device_mdspan_type,
+          std::enable_if_t<is_device_mdspan_v<device_mdspan_type>>* = nullptr>
+auto flatten(device_mdspan_type d_mds)
+{
+  RAFT_EXPECTS(d_mds.is_contiguous(), "Input must be contiguous.");
+  return make_device_vector_view<typename device_mdspan_type::element_type,
+                                 typename device_mdspan_type::layout_type>(d_mds.data(),
+                                                                           d_mds.size());
+}
+
+template <typename mdarray_type, std::enable_if_t<is_mdarray_v<mdarray_type>>* = nullptr>
+auto flatten(const mdarray_type& mda)
+{
+  return flatten(mda.view());
 }
+
+template <typename ElementType, typename LayoutType>
+constexpr auto flatten(host_vector_view<ElementType, LayoutType> h_vv)
```

Comment:
I only added these overloads for a compile-time no-op when `flatten` is called with a `vector` or a `scalar`. However, since the code is not performance critical, I could simplify this and just keep the generic version.